feature/experimental/system/gnmi/benchmarking/ate_tests/drained_configuration_convergence_time/drained_configuration_convergence_time_isis_test.go

```diff
@@ -0,0 +1,134 @@
+)
+
+// setISISOverloadBit is used to configure isis overload bit to true.
+func setISISOverloadBit(t *testing.T, dut *ondatra.DUTDevice) {
+
+	// ISIS Configs to set Overload Bit to true.
+	dutISISPath := gnmi.OC().NetworkInstance(*deviations.DefaultNetworkInstance).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_ISIS, setup.ISISInstance).Isis()
+	lspBit := dutISISPath.Global().LspBit().OverloadBit()
+	gnmi.Replace(t, dut, lspBit.SetBit().Config(), true)
+}
+
+// setISISMetric is used to configure metric on isis interfaces.
+func setISISMetric(t *testing.T, dut *ondatra.DUTDevice) {
+
+	dutISISPath := gnmi.OC().NetworkInstance(*deviations.DefaultNetworkInstance).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_ISIS, setup.ISISInstance).Isis()
+	t.Logf("Configure ISIS metric to %v", setup.ISISMetric)
+	for _, dp := range dut.Ports() {
+		dutISISPathIntfAF := dutISISPath.Interface(dp.Name()).Level(2).Af(oc.IsisTypes_AFI_TYPE_IPV4, oc.IsisTypes_SAFI_TYPE_UNICAST)
+		gnmi.Replace(t, dut, dutISISPathIntfAF.Metric().Config(), setup.ISISMetric)
+	}
+}
+
+// verifyISISMetric is used to verify on ATE to see how much time it
+// has taken to apply changes in metric
+func verifyISISMetric(t *testing.T, dut *ondatra.DUTDevice, ate *ondatra.ATEDevice) {
+
+	t.Run("ISIS Metric verification", func(t *testing.T) {
+		at := gnmi.OC()
+		for _, ap := range ate.Ports() {
+			if ap.ID() == "port1" {
+				// Port1 is ingress, skip verification on ingress port
+				continue
+			}
+
+			const want = oc.Interface_OperStatus_UP
+
+			if got := gnmi.Get(t, ate, at.Interface(ap.Name()).OperStatus().State()); got != want {
+				t.Errorf("%s oper-status got %v, want %v", ap, got, want)
+			}
+			is := at.NetworkInstance(ap.Name()).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_ISIS, "0").Isis()
+			lsps := is.LevelAny().LspAny()
+			gotIsisMetric := gnmi.GetAll(t, ate, lsps.Tlv(oc.IsisLsdbTypes_ISIS_TLV_TYPE_EXTENDED_IPV4_REACHABILITY).ExtendedIpv4Reachability().PrefixAny().Metric().State())
+
+			if diff := cmp.Diff(setup.ISISMetricList, gotIsisMetric); diff != "" {
+				t.Errorf("obtained Metric on ATE is not as expected, got %v, want %v", gotIsisMetric, setup.ISISMetricList)
+			}
+		}
+	})
+}
+
+// verifyISISOverloadBit is used to verify on ATE to see how much time it
+// has taken to apply overload bit on isis adjacencies.
+func verifyISISOverloadBit(t *testing.T, dut *ondatra.DUTDevice, ate *ondatra.ATEDevice) {
+
+	t.Run("ISIS Overload bit verification", func(t *testing.T) {
+		at := gnmi.OC()
+		for _, ap := range ate.Ports() {
+			if ap.ID() == "port1" {
+				// port1 is ingress, skip verification on ingress port
+				continue
+			}
+
+			const want = oc.Interface_OperStatus_UP
+
+			if got := gnmi.Get(t, ate, at.Interface(ap.Name()).OperStatus().State()); got != want {
+				t.Errorf("%s oper-status got %v, want %v", ap, got, want)
+			}
+			// TODO: SetBit is not working on Ixia.
+			// Below code will be uncommented once Ixia issue is resolved.
```

Comment:
Can you please also reference the issue where this support for IXIA is requested?